aidial_adapter_openai/utils/auth.py

```diff
@@ -1,7 +1,41 @@
+import time
 from typing import Mapping, Optional
 
+from azure.core.credentials import AccessToken
+from azure.identity import DefaultAzureCredential
+from fastapi import Request
+from openai import util
+from openai.util import ApiType
 from pydantic import BaseModel
 
+default_credential = DefaultAzureCredential()
+access_token: AccessToken | None = None
+
+EXPIRATION_WINDOW_IN_SEC = 10
+
+
+async def get_api_key() -> str:
+    now = int(time.time()) - EXPIRATION_WINDOW_IN_SEC
+    global access_token
+    if access_token is None or now > access_token.expires_on:
+        access_token = default_credential.get_token(
```

Comment:
need to handle the case when we don't have azure auth variables

Reply:
fixed